transportable-udfs-annotation-processor/src/test/java/com/linkedin/transport/processor/TransportProcessorTest.java

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright 2019 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.transport.processor;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import javax.tools.JavaFileObject;
+import javax.tools.StandardLocation;
+import org.testng.annotations.Test;
+
+import static com.google.testing.compile.JavaFileObjects.*;
+import static com.google.testing.compile.JavaSourcesSubject.*;
+
+
+/**
+ * Tests the {@link TransportProcessor}
+ */
+public class TransportProcessorTest {
+
+  @Test
+  public void simpleUDF() throws IOException {
+    assertThat(
+        forResource("udfs/SimpleUDF.java")
+    ).processedWith(new TransportProcessor())
+        .compilesWithoutError()
+        .and()
+        .generatesFileNamed(StandardLocation.CLASS_OUTPUT, "", Constants.UDF_RESOURCE_FILE_PATH)
+        .withStringContents(Charset.defaultCharset(), asString(forResource("outputs/simpleUDF.json")));
+  }
+
+  @Test
+  public void overloadedUDF() throws IOException {
+    assertThat(
+        forResource("udfs/UDFInterface1.java"),
+        forResource("udfs/OverloadedUDFInt.java"),
+        forResource("udfs/OverloadedUDFString.java")
+    ).processedWith(new TransportProcessor())
+        .compilesWithoutError()
+        .and()
+        .generatesFileNamed(StandardLocation.CLASS_OUTPUT, "", Constants.UDF_RESOURCE_FILE_PATH)
+        .withStringContents(Charset.defaultCharset(), asString(forResource("outputs/overloadedUDF.json")));
+  }
+
+  @Test
+  public void udfWithMultipleInterfaces() throws IOException {
+    assertThat(
+        forResource("udfs/UDFInterface1.java"),
+        forResource("udfs/UDFInterface2.java"),
+        forResource("udfs/UDFWithMultipleInterfaces1.java"),
+        forResource("udfs/AbstractUDF.java"),
+        forResource("udfs/UDFWithMultipleInterfaces2.java")
+    ).processedWith(new TransportProcessor())
+        .compilesWithoutError()
+        .withWarningCount(2)
+        .withWarningContaining(Constants.MULTIPLE_INTERFACES_WARNING)
+        .in(forResource("udfs/UDFWithMultipleInterfaces1.java"))
+        .onLine(14)
+        .atColumn(8)
+        .and()
+        .withWarningContaining(Constants.MULTIPLE_INTERFACES_WARNING)
+        .in(forResource("udfs/UDFWithMultipleInterfaces2.java"))
+        .onLine(14)
+        .atColumn(8)
+        .and()
```

Comment:
redundant?

Reply:
I meant the two 'ands()'

Reply:
Ahh! Yes that is something weird with the way the fluent API is designed. I had to add the two ands. The return types are different, 1st returns `SuccessfulCompilationClause`, 2nd returns `GeneratedPredicateClause<SuccessfulCompilationClause>` which we need.

Reply:
I think this is weird. Probably we should refactor or use the API in a different way?

Reply:
Refactored and split into two test methods